### ISSUE TITLE
Auto terminate quotation after period adds extra quote

### ISSUE THREAD

**Issue by redtopia** [KamasamaK/vscode-cfml#37]
When ending the quotation of a string that ends with a period, an extra quotation character is added as if a new string is being created. See below:

![screenshot](https://user-images.githubusercontent.com/812777/59245690-0636e080-8bd7-11e9-9bc8-a31ade4f95a5.gif)

**Comment by KamasamaK** [comment]
This is automatically handled by VS Code when providing `"` as an [auto-closing pair](https://github.com/KamasamaK/vscode-cfml/blob/4643e0f1c141f5077094a12f7ed67fc06cae3808/language-configuration.json#L15). This could probably be avoided by indicating that this character should not be auto-closed in a string, but for CFML it's useful to do this since that is how quotes are escaped. Is there a reason that you are deleting the quote?

**Comment by redtopia** [comment]
Deleting a quote seems to happen at various times throughout the day for various reasons. But it happens often enough where I find it problematic.

**Comment by redtopia** [comment]
Can I tweak this behavior outside of the CFML extension?

**Comment by KamasamaK** [comment]
You can try adding the following to your settings
`"editor.autoClosingQuotes": "never"`

**Comment by redtopia** [comment]
Thanks... I’ll give that a try, but I do like when the auto-close. What I don’t like is when they auto-close when they shouldn’t. Is there anything in your extension that modifies the auto-close behavior?

**Comment by redtopia** [comment]
In my settings, `Auto Closing Quotes` is set to `languageDefined`. Are there any behaviors that can be modified via the CFML extension? I really don't want to turn it off globally because I do like the feature, and it seems like the behavior I've identified in this issue could be improved.

**Comment by KamasamaK** [comment]
I linked the part that defines the auto-closing behavior above. From there, VS Code handles it.

**Comment by redtopia** [comment]
I made the change you recommended and it works correctly now.

FYI, this was broken in one of the first versions of the Sublime Text 3 extension (https://github.com/jcberquist/sublimetext-cfml), and was fixed. See last screenshot for how it currently works in ST3. As far as allowing a double quote to be auto-completed within a string, I don't believe it would be everyone's preference to preserve that behavior at the expense of this issue. I took a 3 day class with Ortus where I watched the instructor struggle with auto-closing quotes due to this exact issue. At the very least, if you want to preserve the current functionality, perhaps there is a way to make this a user setting?

Prior to change:
![screenshot](https://user-images.githubusercontent.com/812777/59382102-258a5680-8d1a-11e9-877c-6f478e3a9846.gif)

After change:
![screenshot2](https://user-images.githubusercontent.com/812777/59382112-2a4f0a80-8d1a-11e9-9286-f9c1dd24d313.gif)

This is how the Sublime Text 3 CFML plugin works:
![screenshot3](https://user-images.githubusercontent.com/812777/59382395-d7298780-8d1a-11e9-8dce-37ebff1deb46.gif)

**Comment by redtopia** [comment]
For the record, this is what I changed in `language-configuration.json`:

    "autoClosingPairs": [
        ["{", "}"],
        ["[", "]"],
        ["(", ")"],
        { "open": "\"", "close": "\"", "notIn": ["string"] },
        { "open": "'", "close": "'", "notIn": ["string"] },
        ["#", "#"]
    ],

**Comment by KamasamaK** [comment]
I do not believe this is able to be a dynamic or user setting for language configuration. There may be an open issue for VS Code about this, but I could not find it. I can see why this may be an annoyance, but I wouldn't call it "broken" or "incorrect". I'm just not sure how it's a common occurrence that the closing quote is removed.

It's certainly possible to implement a custom auto-closing functionality, but I'm just not sure it would be worth doing so and would ideally need a proper parser (as with most missing or buggy features).

**Comment by redtopia** [comment]
Thank you for responding... changing it as you recommended has definitely made my life better! :)
